### PR TITLE
PLT-6450: Migrate installed_outgoing_webhooks.jsx and installed_outgoing_webhook.jsx to be pure and use Redux

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
WIP: Migrated installed_outgoing_webhooks.jsx and installed_outgoing_webhook.jsx to be pure and use Redux
 * Moved ChannelStore dependency from installed_outgoing_webhook.jsx to
   installed_outgoing_webhooks.js and made it available through props
 * Changed matchesFilter to be static method.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/6454
https://github.com/mattermost/mattermost-server/issues/6450

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has redux changes (please link)

### CODE REVIEW

**Comment by jasonblais** [comment]
@sudheerDev Is this WIP? (noticed it in your description)

**Comment by sudheerDev** [comment]
Test cases are to be added But migration of components is done

**Comment by sudheerDev** [comment]
@jasonblais Yes added test cases as well

File: components/integrations/components/installed_outgoing_webhooks/installed_outgoing_webhooks.jsx [hunk 1]
```diff
@@ -5,82 +5,108 @@ import PropTypes from 'prop-types';
...
         super(props);
 
-        this.handleIntegrationChange = this.handleIntegrationChange.bind(this);
-        this.handleUserChange = this.handleUserChange.bind(this);
         this.regenOutgoingWebhookToken = this.regenOutgoingWebhookToken.bind(this);
-        this.deleteOutgoingWebhook = this.deleteOutgoingWebhook.bind(this);
-
-        const teamId = TeamStore.getCurrentId();
+        this.removeOutgoingHook = this.removeOutgoingHook.bind(this);
+        this.outgoingWebhookCompare = this.outgoingWebhookCompare.bind(this);
```
Comment (jwilander) [inline components/integrations/components/installed_outgoing_webhooks/installed_outgoing_webhooks.jsx]: No change needed, but in the future you can remove these binds and use arrow syntax for the class functions (which auto bind this). A lot of the codebase still has the binds but we're trying to get rid of them as they're extra lines that can be avoided

Comment (sudheerDev) [reply components/integrations/components/installed_outgoing_webhooks/installed_outgoing_webhooks.jsx]: Ya sure. I amended the commit with these changes as well.

**Pull request merged**